Modules/Rpeaks/HilbertTransform.cpp

```diff
@@ -0,0 +1,265 @@
+        y2.push_back(result);
+    }
+
+    vector<double> n(N); //n = 0:N-1
+    iota(n.begin(), n.end(), 0);
+
+    double alpha = 3; //parametr Kaisera, najczęsciej 3
+    vector<double> window(N);
+    for (auto i = 0; i < n.size(); i++) {
+        double ans = pow(2 * n[i] / N - 1, 2);
+        ans = M_PI * alpha * sqrt(1 - ans);
+
+        ans = cyl_bessel_i(0, ans) / cyl_bessel_i(0, M_PI * alpha);
+        window[i] = ans;
+    }
+
+    for (auto i = 0; i < y1.size(); i++) {
+        y1[i] = window[i] * y1[i];
+        y2[i] = window[i] * y2[i];
+    }
+    vector<double> c1 = Conv(y1, signal);
+    Normalize(c1);
+    vector<double> c2 = Conv(y2, c1);
+    Normalize(c2);
+    return c2;
+}
+
+
+double HilbertTransform::CalcRMSValue(vector<double> signal) const {
+    double rms = 0;
+    for (int i = 0; i < signal.size(); i++) {
+        rms += signal[i] * signal[i];
+    }
+    return sqrt(rms / signal.size());
+}
+
+vector<double> HilbertTransform::Derivative(vector<double> signal) const {
+    vector<double> deriv;
+    for (int i = 1; i < signal.size() - 1; i++) { // skip boundaries
+        double acc = 1. / (2 * fs) * (signal[i + 1] - signal[i]);
+        deriv.push_back(acc);
+    }
+    return deriv;
+}
+
+
+vector<double> HilbertTransform::ComputeHilbertTransform(vector<double> signal, int first) const {
+#define REAL(z, i) ((z)[2*(i)])
+#define IMAG(z, i) ((z)[2*(i)+1])
+
+    double data[2 * SIZE];
+
+    for (int i = 0; i < SIZE; i++) {
+        REAL(data, i) = signal[first + i];
+        IMAG(data, i) = 0.0;
+    }
+
+    // fast fourier transform
+    gsl_fft_complex_radix2_forward(data, 1, SIZE);
+
+    // set DC component to 0
+    REAL(data, 0) = 0.0;
+    IMAG(data, 0) = 0.0;
+
+    for (int i = 0; i < SIZE / 2; i++) {
+        gsl_complex a, b;
+        GSL_REAL(a) = REAL(data, i);
+        GSL_IMAG(a) = IMAG(data, i);
+        GSL_REAL(b) = 0.0;
+        GSL_IMAG(b) = -1.0;
+
+        gsl_complex c = gsl_complex_mul(a, b);
+        REAL(data, i) = GSL_REAL(c);
+        IMAG(data, i) = GSL_IMAG(c);
+    }
+
+
+    for (int i = SIZE / 2; i < SIZE; i++) {
+        gsl_complex a, b;
+        GSL_REAL(a) = REAL(data, i);
+        GSL_IMAG(a) = IMAG(data, i);
+        GSL_REAL(b) = 0.0;
+        GSL_IMAG(b) = 1.0;
+
+        gsl_complex c = gsl_complex_mul(a, b);
+        REAL(data, i) = GSL_REAL(c);
+        IMAG(data, i) = GSL_IMAG(c);
+    }
+
+    // inverse fast fourier transform
+    gsl_fft_complex_radix2_inverse(data, 1, SIZE);
+
+    vector<double> h;
+    for (int i = 0; i < SIZE; i++) {
+        h.push_back(REAL(data, i));
+    }
+
+#undef REAL
+#undef IMAG
+    return h;
+}
+
+
+int HilbertTransform::CalcAverageDistance(vector<int> peaks) const {
+    double distance = 0;
+    for (int i = 1; i < peaks.size(); i++) {
+        distance += (peaks[i] - peaks[i - 1]);
+    }
+    return distance / (peaks.size() - 1);
+}
+
+
+vector<int> HilbertTransform::GetPeaks(vector<double> electrocardiogram_signal) const {
```

Comment:
W nazwach zmiennych stosujemy `camelCase` a nie `snake_case`. Zastosuje to proszę do wszystkich wystąpień

Reply:
Najpierw miałem wszystkie `camelCase`, ale przeczytałem plik podesłany przez Bartka: 
https://google.github.io/styleguide/cppguide.html#Variable_Names
I zmieniałem część na `snake_case`. W tej konwencji pisze się: `const int constVariable`, `int normal_variable` dla odróżnienia tych dwóch. Trochę szkoda to zamieniać, bo ułatwia to czytanie kodu.

Reply:
Ten wytyczne nie do końca są dobre według mnie ale niech zostanie.

Reply:
Te wytyczne nie są do końca dobre według mnie. Jeśli tak jest dla Ciebie czytelniej, niech zostanie.

Reply:
To zostawiam